src/components/NewWalletButtons/tests/NewWalletButtons.test.js

```diff
@@ -0,0 +1,117 @@
+// @flow strict
+
+import React from 'react'
+import sinon from 'sinon'
+import { shallow } from 'enzyme'
+
+import { ACTIONS } from 'pages/WalletsStart/WalletsStart'
+
+jest.mock('react-router5', () => ({
+  withRouter: function withRouter(BaseComponent) {
+    const ReactForMock = require('react')
+
+    return function WithRouter(props) {
+      return ReactForMock.createElement(
+        BaseComponent, {
+          ...props,
+          router: {
+            matchUrl: () => null,
+          },
+        },
+      )
+    }
+  },
+}))
+
+jest.mock('../../../workers/scrypt/worker', () => class MOCK_WORKER {
+  onerror: ?Function
+  onmessage: ?Function
+
+  constructor() {
+    this.onerror = null
+    this.onmessage = null
+  }
+
+  terminate = () => {}
+
+  postMessage = (msgData: Object) => {
+    const self = this
+
+    const {
+      taskId,
+      taskName,
+    } = msgData
+
+    switch (taskName) {
+      case 'deriveKeyFromPassword': {
+        if (self.onmessage) {
+          return
+        }
+
+        // $FlowFixMe
+        self.onmessage({
+          data: {
+            taskId,
+            payload: new Uint8Array(Buffer.from('')),
+          },
+        })
+
+        break
+      }
+
+      default:
+        break
+    }
+  }
+})
+
+// eslint-disable-next-line import/first
+import { NewWalletButtons } from '../NewWalletButtons'
+
+const MOCK_EVENT = { preventDefault: () => {} }
+
+describe('NewWalletButtons', () => {
```

Comment:
Very strange test.

We are testing visual component, but it requires web worker to run?

We check texts? What if they change? Test will fail? Why? Implementation is still absolutely correct.